src/main/java/jarvis/model/TaskDeadline.java

```diff
@@ -34,11 +34,12 @@ public String toString() {
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof TaskDeadline // instanceof handles nulls
+                && deadline != null && ((TaskDeadline) other).deadline != null
```

Comment:
I think instead of this we can require `deadline` to be non-null in the constructor.

Reply:
The null check here is added because deadline is an optional field for tasks and for ease of implementation I used null in the TaskDeadline class to represent an empty TaskDeadline. So this added line is my attempt to implement more null checks to prevent null pointer exceptions because I realised that there could be null pointer exceptions in the following line when the equals method is called.

Regardless I also just realised this implementation is actually wrong in checking that either both deadlines are not null or both are null.

Reply:
Hmm in that case I think it would be better to make `TaskDeadline.deadline` a compulsory non-null field and `Task.taskDeadline` an optional field. This way we don't create a `TaskDeadline` object if the task does not have a deadline.

Reply:
Ok I think that makes sense too, would be easier to understand and more consistent with the other classes. Fixed in my latest commits.